src/LtiGrade.php

```diff
@@ -169,4 +174,27 @@ public function setCanvasExtension($value)
 
         return $this;
     }
+
+    /**
+     * Validates that the given time is an ISO 8601 string with sub-second precision
+     * per the LTI spec. Some LMS (e.g. Schoology) include validation logic that will
+     * throw a validation error if this isn't enforced.
+     *
+     * @see https://www.imsglobal.org/spec/lti-ags/v2p0#timestamp
+     * @throws InvalidArgumentException Thrown if the given timestamp is not an ISO8601
+     *   compatible string with sub-second precision.
+     */
+    private function validateTimestamp(string $timestamp): void
+    {
+        // Attempt to parse the timestamp as an ISO8601 string.
+        $dt = DateTime::createFromFormat(DateTime::ATOM, $timestamp);
```

Comment:
Note: This needs to be double-checked. ISO 8601 is a surprisingly permissive format.